generators/server/index.js

```diff
@@ -278,6 +282,46 @@ module.exports = JhipsterServerGenerator.extend({
                 },
                 {
                     when: function (response) {
+                        return applicationType === 'gateway' || applicationType == 'microservice';
+                    },
+                    type: 'list',
+                    name: 'authenticationType',
+                    message: function (response) {
+                        return getNumberedQuestion('Which *type* of authentication would you like to use?', currentQuestion, totalQuestions, function (current) {
+                            currentQuestion = current;
+                        }, applicationType === 'monolith');
+                    },
+                    choices: [
+                        {
+                            value: 'jwt',
+                            name: 'JWT authentication (stateless, with a token)'
+                        },
+                        {
+                            value: 'uaa',
+                            name: 'Authentication via JHipster UAA (you will have to generate one separately)'
+                        }
+                    ],
+                    default: 0
+                },
+                {
+                    when: function (response) {
+                        return (applicationType === 'gateway' && response.authenticationType === 'uaa');
+                    },
+                    type: 'input',
+                    name: 'uaaBaseName',
+                    validate: function (input) {
+                        if (true) return true; //@TODO check it really
+                        return 'Could not finde UAA application "' + input + '"';
+                    },
+                    message: function (response) {
+                        return getNumberedQuestion('What is the path of your UAA application?.', currentQuestion, totalQuestions, function (current) {
```

Comment:
how will you know the folder path? if you are asking folder Path here then how will you know the uaa app name? say for example I can choose a baseName for uaa app like `myAwesomeUaa` and my app can be in a folder `../security/uaaApp` in this case how will you get the app name? are you goann read the `yo-rc.json` in the folder and read base name from that? I guess that would be the only way